[rollingpencil] iP

rollingpencil opened the pull request:
# Tohru

> D! D is for **dragon**. :dragon:

Tohru is your personal task assistant. It's:

- text-based _(currently)_
- ad-free
- ~offer roasted tail for dinner~

## Installation:

1. Download the jar from [releases](https://github.com/rollingpencil/ip/releases/tag/A-Jar).
2. Copy the Java archive to a folder of your choice.
3. Open terminal and run:

```bash
java -jar Tohru-<version>.jar
```

## Usage:
Available commands are:

`list`, `todo`, `deadline`, `event`, `bye`, `mark`, `unmark`

## Roadmap:

- [x] CLI client
- [ ] GUI Client

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

rollingpencil committed:
Implement initial skeletal greet

rollingpencil committed:
Update name of chatbot

rollingpencil committed:
Amend chatbot to echo user input

rollingpencil committed:
Add TodoItem class

rollingpencil committed:
Add TodoList class

rollingpencil committed:
Update chatbot to add and list entries in todolist

rollingpencil committed:
Modify TodoItem to track completed status

rollingpencil committed:
Modify TodoList to support editing of TodoItems

rollingpencil committed:
Add support for mark and unmark commands

rollingpencil committed:
Update the string representation of TodoItem

rollingpencil committed:
Add DeadlineItem class

rollingpencil committed:
Add EventItem class

rollingpencil committed:
Change addItem method to accept TodoItem object instead of string

rollingpencil committed:
TodoList.java: Add getTotal method to get count of entries

rollingpencil committed:
Update chatbot to support todo, deadline and event commands

rollingpencil committed:
Add JavaDoc comments for DeadlineItem, EventItem classes

rollingpencil committed:
Modify todo, deadline, event commands to trim task content for output

rollingpencil committed:
runtest.sh: Make script executable and update chatbot name in script

rollingpencil committed:
Update test input and test expected output

rollingpencil committed:
Add Exception

rollingpencil committed:
Handle user input errors

rollingpencil committed:
Refactor TodoList to use ArrayList

rollingpencil committed:
TodoList.java: Add deleteItem method

rollingpencil committed:
Add delete command to chatbot

rollingpencil committed:
Tohru.java: Update indentation for switch case

rollingpencil committed:
Tohru.java: Update deadline argument validation

rollingpencil committed:
Add FileStore to handle save related functions

rollingpencil committed:
TodoItem.java, DeadlineItem.java, EventItem.java: Add getSaveString()

rollingpencil committed:
TodoList.java: Integrate FileStore

rollingpencil committed:
Merge branch 'branch-Level-7'

rollingpencil committed:
DeadlineItem.java: Update deadline field to be LocalDateTime type

rollingpencil committed:
Tohru.java: Update deadline command to parse datetime string

rollingpencil committed:
FileStore.java: Update retrieveTodoList() to handle datetime strings

rollingpencil committed:
EventItem.java: Update from and to fields to be LocalDateTime type

rollingpencil committed:
FileStore.java: Update retrieveTodoList() to handle datetime for events

rollingpencil committed:
Tohru.java: Update event command to handle datetime strings

rollingpencil committed:
Update test to include new datetime format

rollingpencil committed:
Merge branch 'branch-Level-8'

rollingpencil committed:
Add Command.java

rollingpencil committed:
Add Ui.java

rollingpencil committed:
Add Parser.java and update Command.java structure

rollingpencil committed:
Add ByeCommand.java

rollingpencil committed:
Add ListCommand.java

rollingpencil committed:
Decouple FileStore, List Command and TodoList

rollingpencil committed:
Add MarkCommand.java

rollingpencil committed:
Add UnmarkCommand.java

rollingpencil committed:
Add DeleteCommand.java

rollingpencil committed:
Add AddTodoCommand.java

rollingpencil committed:
Add AddDeadlineCommand.java

rollingpencil committed:
Add AddEventCommand.java

rollingpencil committed:
Ui.java: Implement user input methods

rollingpencil committed:
ListCommand.java: Remove unnecessary imports

rollingpencil committed:
Tohru.java: Refactor control flow to utilise Parser and Command classes

rollingpencil committed:
TodoList.java: Update JavaDoc on methods

rollingpencil committed:
Merge branch 'branch-A-MoreOOP'

rollingpencil committed:
Refactor into packages

rollingpencil committed:
Merge branch 'branch-A-Packages'

rollingpencil committed:
Merge branch 'add-gradle-support'

rollingpencil committed:
build.gradle: Update mainClass

rollingpencil committed:
Add ParserTest.java

rollingpencil committed:
Add FileStoreStub.java to stub FileStore for testing purposes

rollingpencil committed:
Add AddDeadlineCommandTest.java

rollingpencil committed:
Add AddEventCommandTest.java

rollingpencil committed:
Merge branch 'branch-A-JUnit'

rollingpencil committed:
build.gradle: Add archiveVersion

rollingpencil committed:
Update JavaDocs

rollingpencil committed:
Command.java: Update JavaDoc

rollingpencil committed:
Merge branch 'branch-A-JavaDoc'

rollingpencil committed:
DeadlineItem.java: Fix Typo

rollingpencil committed:
Merge branch 'branch-A-CodingStandard'

rollingpencil committed:
Implement find command

rollingpencil committed:
Refactor FileStore

rollingpencil committed:
Merge branch 'branch-Level-9'

rollingpencil committed:
Fix single line JavaDoc formatting

rollingpencil committed:
Add checkstyle configuration files

rollingpencil committed:
build.gradle: Configure checkstyle

rollingpencil committed:
Fix import order

rollingpencil committed:
Merge branch 'branch-A-CheckStyle'

rollingpencil committed:
build.gradle: Add JavaFX dependencies

rollingpencil committed:
Add Ui interface and refactor existing Ui to Tui

rollingpencil committed:
Add MainWindow and DialogBox

avrilmohh commented on src/main/java/tohru/command/Command.java:
```diff
@@ -0,0 +1,49 @@
+ */
+public abstract class Command {
+
+    /** Formatter used to parse datetime strings in arguments */
```
```suggestion
    /** Formatter used to parse datetime strings in arguments. */
```

avrilmohh commented on src/main/java/tohru/command/Command.java:
```diff
@@ -0,0 +1,49 @@
+    /** Formatter used to parse datetime strings in arguments */
+    protected static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("d/M/y HHmm");
+
+    /** The argument passed in to a command */
```
```suggestion
    /** The argument passed in to a command. */
```

avrilmohh commented on src/main/java/tohru/command/ByeCommand.java:
```diff
@@ -0,0 +1,40 @@
+ */
+public class ByeCommand extends Command {
+
+    /** Prefix used to invoke the bye command */
```
```suggestion
    /** Prefix used to invoke the bye command. */
```

avrilmohh commented on src/main/java/tohru/command/AddTodoCommand.java:
```diff
@@ -0,0 +1,47 @@
+ */
+public class AddTodoCommand extends Command {
+
+    /** Prefix used to invoke the add to-do command */
```
```suggestion
    /** Prefix used to invoke the add to-do command. */
```

avrilmohh commented on src/main/java/tohru/command/AddEventCommand.java:
```diff
@@ -0,0 +1,80 @@
+ */
+public class AddEventCommand extends Command {
+
+    /** Prefix used to invoke the add deadline command */
```
```suggestion
    /** Prefix used to invoke the add deadline command. */
```

avrilmohh commented on src/main/java/tohru/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,68 @@
+ */
+public class AddDeadlineCommand extends Command {
+
+    /** Prefix used to invoke the add deadline command */
```
```suggestion
    /** Prefix used to invoke the add deadline command. */
```

avrilmohh commented on src/main/java/tohru/command/DeleteCommand.java:
```diff
@@ -0,0 +1,54 @@
+ */
+public class DeleteCommand extends Command {
+
+    /** Prefix used to invoke the delete command */
```
```suggestion
    /** Prefix used to invoke the delete command. */
 ```

avrilmohh commented on src/main/java/tohru/command/FindCommand.java:
```diff
@@ -0,0 +1,47 @@
+ */
+public class FindCommand extends Command {
+
+    /** Prefix used to invoke the find command */
```
```suggestion
    /** Prefix used to invoke the find command. */
```

avrilmohh commented on src/main/java/tohru/command/ListCommand.java:
```diff
@@ -0,0 +1,36 @@
+ */
+public class ListCommand extends Command {
+
+    /** Prefix used to invoke the list command */
```
```suggestion
    /** Prefix used to invoke the list command. */
```

avrilmohh commented on src/main/java/tohru/command/MarkCommand.java:
```diff
@@ -0,0 +1,52 @@
+ */
+public class MarkCommand extends Command {
+
+    /** Prefix used to invoke the mark command */
```
```suggestion
    /** Prefix used to invoke the mark command. */
```

avrilmohh commented on src/main/java/tohru/command/Parser.java:
```diff
@@ -0,0 +1,72 @@
+import tohru.exception.TohruException;
+
+/**
+ * Parser represents the parsing of input prompt and determine which command to execute
```
```suggestion
 * Parser represents the parsing of input prompt and determine which command to execute.
```

avrilmohh commented on src/main/java/tohru/command/Parser.java:
```diff
@@ -0,0 +1,72 @@
+public class Parser {
+
+    /**
+     * Parse the giving input and determine the corresponding commands
+     *
+     * @param prompt The input from the user
+     * @return A command object representing the operation requested by the user
+     * @throws TohruException When an invalid command is passed in
+     */
```
```suggestion
     * Parse the giving input and determine the corresponding commands.
     *
     * @param prompt The input from the user.
     * @return A command object representing the operation requested by the user.
     * @throws TohruException When an invalid command is passed in.
     */
```

avrilmohh commented on src/main/java/tohru/command/Parser.java:
```diff
@@ -0,0 +1,72 @@
+
+    }
+
+    /**
+     * Helper function to parse the datetime string passed in with the provided formatter
+     *
+     * @param datetimeString The datetime string to be parsed
+     * @param formatter The format used to parse the datetime string
+     * @return A LocalDateTime object with the parsed date and time
+     * @throws TohruException When the datetime string does not follow the format specified
+     */
```
```suggestion
    /**
     * Helper function to parse the datetime string passed in with the provided formatter.
     *
     * @param datetimeString The datetime string to be parsed.
     * @param formatter The format used to parse the datetime string.
     * @return A LocalDateTime object with the parsed date and time.
     * @throws TohruException When the datetime string does not follow the format specified.
     */
```

avrilmohh commented on src/main/java/tohru/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,53 @@
+ */
+public class UnmarkCommand extends Command {
+
+    /** Prefix used to invoke the mark command */
```
```suggestion
    /** Prefix used to invoke the mark command. */
```

avrilmohh commented on src/main/java/tohru/storage/FileStore.java:
```diff
@@ -0,0 +1,199 @@
+ */
+public class FileStore {
+
+    /** Represents the savefile on local storage */
```
```suggestion
    /** Represents the savefile on local storage. */
```

avrilmohh commented on src/main/java/tohru/storage/FileStore.java:
```diff
@@ -0,0 +1,199 @@
+            return todoList;
+        }
+
+        int errorEntriesCount = 0; //Counter to count the number of error entries
```
```suggestion
        int errorEntriesCount = 0; // Counter to count the number of error entries
```

avrilmohh commented on src/test/java/tohru/storage/FileStoreStub.java:
```diff
@@ -0,0 +1,25 @@
+
+import tohru.task.TodoItem;
+
+/**
+ * A stub class that represent hte FileStore class simulating loading
+ * and saving from local storage
+ */
```
```suggestion
/**
 * A stub class that represent hte FileStore class simulating loading.
 * and saving from local storage.
 */
```

avrilmohh reviewed (commented):
Amazing job! I found your code easy to read because of the extensive Javadoc comments and normal comments for each step of your code. However, there are some minor coding standard violations in the Javadoc comments, where some comments have a full stop at the end while some didn't, so do remember to add a full stop after your code comments (according to the course coding standard).

rollingpencil committed:
Ui.java: Make showText and showError parameter varargs

rollingpencil committed:
Fix JavaDoc comments

rollingpencil committed:
Add CSS to user interface

rollingpencil committed:
Add delay to show goodbye message before exit

Brendan8899 commented on src/main/java/tohru/command/Parser.java:
```diff
@@ -0,0 +1,72 @@
+            arguments = dissectedPrompt[1].trim();
+        }
+
+        switch (command) {
```
Nice Egyptian Brace and Good Indentation for Switch Case. Keep up the good work.

rollingpencil committed:
Merge branch 'branch-Level-10'

Brendan8899 reviewed (commented):
Looks Great to Me

rollingpencil committed:
Update JavaDoc

rollingpencil committed:
Fix typo in CSS

rollingpencil committed:
Merge branch 'branch-A-Varargs'

CreditableX commented on src/main/java/tohru/exception/TohruException.java:
```diff
@@ -0,0 +1,16 @@
+package tohru.exception;
```
TohruException could be split into a few further exceptions classes to specify what type of error? E.g TohruStringNotFound

CreditableX commented on src/test/java/tohru/command/AddDeadlineCommandTest.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    @Test
+    public void execute_correctFormatArguments_success() {
```
Nice test class naming!

CreditableX commented on src/main/java/tohru/storage/FileStore.java:
```diff
@@ -0,0 +1,199 @@
+                    errorEntriesCount++;
+                    continue;
+                }
+                boolean status = Boolean.parseBoolean(entryStatusString);
```
Maybe status could be changed to isComplete or something along those lines?

CreditableX commented on src/main/java/tohru/task/TodoList.java:
```diff
@@ -0,0 +1,121 @@
+     *
+     * @return The number of entries in the list.
+     */
+    public int getTotal() {
```
Maybe getTotalItems would be better?

rollingpencil committed:
Add assertion to various methods

The data classes and exception classes assumes the arguments passed in
are valid. The validity is assumed the arguments passed to be not null
or blank.

These assertion act as a secondary checks for the various data classes
as the guard clauses should have caught such errors parsing the input.

Let's add null assertion and empty assertion checks to
* Data classes such as TodoItem, EventItem, DeadlineItem and TodoList.
* Exception classes such as TohruException and CorruptSaveException.

rollingpencil committed:
Refactor Command classes

The MarkCommand, UnmarkCommand and DeleteCommand classes has duplicate
code chunks that deals with parsing the argument passed in as an
integer. The ListCommand and FindCommand classes has duplicate code
chunks that deals with converting to-do entries into itemised string
format. Additionally, The AddDeadlineCommand and AddEventCommand
classes does quite a few checks on data validity, which makes the
method feel bloated.

These duplicated code makes it a hassle to update the logic for the
argument parsing in each file for the above Command classes.

Let's refactor
* MarkCommand, UnmarkCommand and DeleteCommand classes to use a
common integer parser.
* ListCommand and FindCommand classes to use a common
getItemisedItemsString() method to get the itemised string
representation.
* AddDeadlineCommand and AddEventCommand to use a separate
method to handle the parsing of arguments.

rollingpencil committed:
Refactor FileStore class

The retrieveTodoList() method is very bloated as it handles:
* reading from file
* error handling
* creation of TodoItem, DeadlineItem and EventItem
* argument validation
* determination of task type

Due to the large responsibilities handled by the method, it makes the
readability of method very poor.

Let's refactor the method to:
* handle the reading of save file from file system
* create a separate method to parse each entry in the save file
* create a separate parsers for each of the task types.

rollingpencil committed:
Merge pull request #1 from rollingpencil/branch-A-Assertions

Increment: A-Assertions

rollingpencil committed:
Merge branch 'master' into branch-A-CodeQuality

rollingpencil committed:
Merge pull request #2 from rollingpencil/branch-A-CodeQuality

Increment: A-CodeQuality

rollingpencil committed:
Add GitHub workflow gradle.yml

rollingpencil committed:
Update trigger events

rollingpencil committed:
Merge pull request #3 from rollingpencil/branch-A-CI

Increment: A-CI

rollingpencil committed:
Modify find command to do fuzzy search

rollingpencil committed:
Merge pull request #4 from rollingpencil/branch-BCD-Extension

Increment: C-BetterSearch

rollingpencil committed:
TodoItem.java: Change name of completion status

rollingpencil committed:
Update README to show command usage

rollingpencil committed:
README: Update acknowledgements

rollingpencil committed:
Merge pull request #6 from rollingpencil/branch-A-UserGuide

Add User guide

rollingpencil committed:
build.gradle: Update archiveVersion number
